invoicing/api/views.py

```diff
@@ -59,3 +66,48 @@ class ApartmentInstallmentAPIView(InstallmentAPIViewBase):
     queryset = ApartmentInstallment.objects.all()
     serializer_class = ApartmentInstallmentSerializer
     parent_field = "apartment_reservation_id"
+
+
+@extend_schema(
+    description="Create an invoice PDF based on apartment installments.",
+    parameters=[
+        OpenApiParameter(
+            name="index",
+            description="Comma-separated row index numbers starting from 0.",
+            type={"type": "array", "items": {"type": "number"}},
+            location=OpenApiParameter.QUERY,
+            required=False,
+        )
+    ],
+    responses={(200, "application/pdf"): OpenApiTypes.BINARY},
+)
+class ApartmentInstallmentInvoiceAPIView(APIView):
+    def get(self, request, **kwargs):
+        installments = list(
+            ApartmentInstallment.objects.filter(
+                apartment_reservation_id=kwargs["apartment_reservation_id"]
+            ).order_by("id")
+        )
+
+        if index_params := request.query_params.get("index"):
+            installments = [
+                _find_installment_by_index_param(index_param, installments)
+                for index_param in index_params.split(",")
+            ]
+
+        pdf_data = create_invoice_pdf_from_installments(installments)
+        response = HttpResponse(pdf_data, content_type="application/pdf")
+        response["Content-Disposition"] = f"attachment; filename={INVOICE_FILE_NAME}"
```

Comment:
This is an opinion matter. I hate to get same name files that replace the existing ones in the `downloads` folder. 😄 Can here be some separator in filename ex. apartment_number + INVOICE_FILE_NAME or some other?